Fix a typo (#5544)
MicrosoftDocs/visualstudio-docs: docs/code-quality/ca2007.md

## When to suppress warnings

> You can suppress this warning if you know that the consumer is a graphical user interface (GUI) app or if the consumer does not have a <xref:System.Threading.SynchronizationContext>.

Author: @mavasani, I'm not sure about the other part of the sentence related to the SynchronizationContext

Reviewer: Tagging @stephentoub. https://devblogs.microsoft.com/dotnet/configureawait-faq/ for reference.

Reviewer 2: Thanks.  I'd prefer to see it more like this:
```suggestion
This warning is intended for libraries, where the code may be executed in arbitrary environments and where code shouldn't make assumptions about the environment or how the caller of the method may be invoking or waiting on it.  It is generally appropriate to suppress / disable the warning entirely over projects that represent application code rather than library code; in fact, running this analyzer on application code (e.g. button click event handlers in a Windows Forms project) is likely to lead to the wrong actions being taken.

You can suppress this warning in any situation where either the continuation should be scheduled back to the original context or where there is no such context in place.  For example, when writing code in a button click event handler in a Windows Forms application, in general the continuation from an await should run on the UI thread, and thus the default behavior of scheduling the continuation back to the originating context is desirable.  As another example, when writing code in an ASP.NET Core application, by default there is no SynchronizationContext or TaskScheduler, such that a `ConfigureAwait` wouldn't actually change any behavior.
```

@mavasani, is CA2007 enabled by default in the new SDK analyzers library?  If yes, we should change that.

Author: @stephentoub Thanks. It's much more detailed now. I've applied the suggestion and added a few small tweaks (mention WPF too, added xref to class names, replaced e.g. with for example based on [this guidelines](https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg).

One question got to my mind now. What if I'm writing a class library that's intended to be used in **my own application only**, and I can **guarantee** that it will **always** be called asynchronously. Does it matter in that case whether I use ConfigureAwait(false) in my class library?

Reviewer 2: > Does it matter in that case whether I use ConfigureAwait(false) in my class library?

No, at that point it's basically application code.

## When to suppress warnings

> This warning is intended for libraries, where the code may be executed in arbitrary environments and where code shouldn't make assumptions about the environment or how the caller of the method may be invoking or waiting on it. It is generally appropriate to suppress / disable the warning entirely over projects that represent application code rather than library code; in fact, running this analyzer on application code (for example, button click event handlers in a WinForms/WPF project) is likely to lead to the wrong actions being taken.

Reviewer 3: ```suggestion
This warning is intended for libraries, where the code may be executed in arbitrary environments and where code shouldn't make assumptions about the environment or how the caller of the method may be invoking or waiting on it. It is generally appropriate to suppress the warning entirely for projects that represent application code rather than library code; in fact, running this analyzer on application code (for example, button click event handlers in a WinForms or WPF project) is likely to lead to the wrong actions being taken.
```

> You can suppress this warning in any situation where either the continuation should be scheduled back to the original context or where there is no such context in place. For example, when writing code in a button click event handler in a WinForms/WPF application, in general the continuation from an await should run on the UI thread, and thus the default behavior of scheduling the continuation back to the originating context is desirable. As another example, when writing code in an ASP.NET Core application, by default there is no <xref:System.Threading.SynchronizationContext> or <xref:System.Threading.Tasks.TaskScheduler>, such that a `ConfigureAwait` wouldn't actually change any behavior.

Reviewer 3: ```suggestion
You can suppress this warning in any situation where either the continuation should be scheduled back to the original context or where there is no such context in place. For example, when writing code in a button click event handler in a WinForms or WPF application, in general the continuation from an await should run on the UI thread, and thus the default behavior of scheduling the continuation back to the originating context is desirable. As another example, when writing code in an ASP.NET Core application, by default there is no <xref:System.Threading.SynchronizationContext> or <xref:System.Threading.Tasks.TaskScheduler>, for which reason a `ConfigureAwait` wouldn't actually change any behavior.
```
